### PR TITLE
[proposal] Edgemesh Container Network supports CNI features

### PR DESCRIPTION
**Pull request by IdeaMeshDyx**
What type of PR is this?
/kind design

What this PR does / why we need it:

This proposal provides a design for edgemesh to support CNI feature.
There are two main goals:

Implement network partitioning and container IP allocation functions, compatible with CNI standards, and ensure that the IP address of each container in the cluster is unique.
Support cross-subnet Pod IP traffic proxy and forwarding functions, compatible with CNI standards.
Which issue(s) this PR fixes:

Fixes https://github.com/kubeedge/edgemesh/issues/465

### CODE REVIEW

**Comment by Poorunga** [comment]
@IdeaMeshDyx Need fix conflict

File: pkg/apis/config/v1alpha1/default.go.bak
Comment (JiaweiGithub) [inline pkg/apis/config/v1alpha1/default.go.bak:1]: 这些.bak文件，确定不用了的，删下吧

File: cmd/edgemesh-agent/app/server.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package app
 
 import (
 	"fmt"
+	"github.com/kubeedge/edgemesh/pkg/cni"
```
Comment (Poorunga) [inline cmd/edgemesh-agent/app/server.go:5]: go import顺序请遵守如下规范：
1. 内置包
2. 第三方包
3. 项目自身的包

用空行间隔

Comment (Poorunga) [reply cmd/edgemesh-agent/app/server.go:5]: 下方所有go文件做相同处理

File: pkg/apis/config/v1alpha1/default.go.bak
Comment (IdeaMeshDyx) [reply pkg/apis/config/v1alpha1/default.go.bak:1]: 好的

File: cmd/edgemesh-agent/app/server.go (hunk 1 above)
Comment (IdeaMeshDyx) [reply cmd/edgemesh-agent/app/server.go:5]: 好的

**Comment by Poorunga** [comment]
CI / Check helm package (pull_request) Failing  一直出错的原因是，你自己生成的edgemsh.tgz 文件和 github自动生成的tgz，md5不一致 @IdeaMeshDyx

**Comment by Poorunga** [comment]
@IdeaMeshDyx 需要执行 ./hack/verify-vendor.sh 和 ./hack/update-vendor.sh 和 ./hack/update-vendor-licenses.sh 去更新vendor和licenses

File: go.mod [hunk 2]
```diff
@@ -37,6 +35,11 @@ require (
 	sigs.k8s.io/yaml v1.2.0
 )
 
+require (
```
Comment (Poorunga) [inline go.mod:38]: 此require和上方require合并

File: pkg/cni/adapter.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
+package cni
+
+import (
+	"context"
+	"fmt"
+	"github.com/kubeedge/edgemesh/pkg/apis/config/defaults"
```
Comment (Poorunga) [inline pkg/cni/adapter.go:6]: import顺序遵循规范

File: pkg/util/cni/tun.go [hunk 4]
```diff
@@ -0,0 +1,367 @@
+package cni
+
+import (
+	"fmt"
+	"github.com/songgao/water"
```
Comment (Poorunga) [inline pkg/util/cni/tun.go:5]: 包导入顺序

**Pull request merged**